### DISCUSSION TITLE
Using other camera models - not regular pinhole

### DISCUSSION THREAD

**Discussion by avispi** [cdcseacave/openMVS#741] [Q&A]
Hello.

First of all- I want to say I was amazed by the results on several scenes I have experience with. Your product is wonderful and it is amazing to have it opensource.

How dependent is the process on the camera models being regular pinhole models?

Is it possible to have a camera model interface, one instance will be the current camera class, another is a distorted pinhole model and maybe other models as well?

I see some uses assuming camera has K R and C. 
I see that in some use cases, C can be replaced by some GetPosition() interface and the uses of R can also be replaced with appropriate interface (i.e to rotate normal from camera space).

Do you think there is some algorithmic assumptions that can not be made for a non-pinhole model?

Thank you.

**Comment by cdcseacave** [comment]
Everything can be made to work with general camera models; however I do not see any reason to do so for distorted pinhole camera models, which can and should be undistorted before MVS to speed it up.

If you have other camera models in mind pls be more specific.

**Reply by avispi** [reply]
Thank you. 
Yes. distorted pinhole was an example.
Practically, not a good one.
But other physical or mathematical models (i.e: RPC ) cannot do that.
To use these model, I think one of the following should be implemented:
1. an api function of stereorectification for  any new camera model
2. A semi global match method that does not assume stereo rectified pair and works with the rpipolar geometry derived from the model 
      (i.e: something like an epipolar curve in case of rpc)
Any thoughts or suggestions?
Thanks again.

**Reply by cdcseacave** [reply]
Sure, unrectified images can be use with a bit of work, but expect some speed drops. Do you have a specific usecase in mind or this is just a theoretical question?

**Reply by avispi** [reply]
Hi.
The main use case I was thinking about is satellite imagery.

**Reply by cdcseacave** [reply]
can be done, however I do not have time for implementing it, we can ask help from the comunity
